roofit/xroofit/src/xRooNode.cxx

```diff
@@ -0,0 +1,6879 @@
+            if (nomf && otherf) {
+                otherf->dataHist().set(*bin_pars, 2*nomf->dataHist().weight(bin-1) - value);
+                otherf->setValueDirty();
+            }
+        }
+        sterilize();
+        return true;
+    } else if (auto f2 = dynamic_cast<RooStats::HistFactory::FlexibleInterpVar*>(o); f2) {
+        // changing nominal value
+        f2->setNominal(value);
+    }
+    throw std::runtime_error(TString::Format("unable to set bin content of %s",GetPath().c_str()));
+
+}
+
+bool xRooNode::SetBinData(int bin, double value, const char* dataName) {
+    return datasets()[dataName]->SetBinContent(bin,value);
+}
+
+bool xRooNode::SetBinError(int bin, double value) {
+
+    // if it's a RooProduct locate child with the same name
+    if (get<RooProduct>()) {
+        return factors()[GetName()]->SetBinError(bin,value);
+    }
+
+    if(auto _varies = variations(); !_varies.empty()) {
+        return _varies["nominal"]->SetBinError(bin,value);
+    }
+
+    auto o = get();
+
+    if (auto f = dynamic_cast<RooHistFunc*>(o); f) {
+
+        //if (f->getAttribute("density")) { value /= f->dataHist().binVolume(*bin_pars); } - commented out because DON'T convert .. sumw and sumw2 attributes will be stored not as densities
+
+        // NOTE: Can only do this because factors() makes parents of its children it's own parent (it isn't the parent)
+        // If ever make factors etc part of the parentage then this would need tweaking to get to the true parent
+        // find first parent that is a RooProduct, that is where the statFactor would live
+        // stop as soon as we reach pdf object
+        auto _prodParent = fParent;
+        while (_prodParent && !_prodParent->get<RooProduct>() && !_prodParent->get<RooAbsPdf>()) {
+            if (_prodParent->get<PiecewiseInterpolation>() && strcmp(GetName(),"nominal")) {
+                _prodParent.reset(); break; // only the 'nominal' variation can look for a statFactor outside the variation container
+            }
+            _prodParent = _prodParent->fParent;
+        }
+        auto _f_stat = (_prodParent && !_prodParent->get<RooAbsPdf>()) ? _prodParent->factors().find("statFactor") : nullptr;
+        auto f_stat = (_f_stat) ? _f_stat->get<ParamHistFunc>() : nullptr;
+        if (_f_stat && _f_stat->get() && !f_stat) {
+            throw std::runtime_error("stat factor must be a paramhistfunc");
+        }
+
+        // stat uncertainty lives in the "statFactor" factor, each sample has its own one,
+        // but they can share parameters
+        if (!f_stat) {
+            if (value==0) return true;
+            TString parNames;
+            for(auto& p : xRooNode("tmp", *f, std::shared_ptr<xRooNode>(nullptr)).vars()) {
+                if (parNames!="") parNames += ",";
+                parNames += p->get()->GetName();
+            }
+            auto h = std::shared_ptr<TH1>( f->dataHist().createHistogram(parNames) );
```

Comment:
```suggestion
            auto h = std::unique_ptr<TH1>( f->dataHist().createHistogram(parNames, RooCmdArg::none()) );
```
I realized now that this is the best way to get rid of the deprecation warning. By passing an empty command argument you are selecting the right overload of createHistogram (the one taking command arguments). This is only a temporary fix, because as long as the deprecated overload is still there, it will be selected when you are not passing any additional arguments.

See also https://github.com/root-project/root/pull/10411.

Reply:
Also, there is no need to use `shared_ptr` there because the object is not shared